address.go

```diff
@@ -161,11 +162,32 @@ func (a Address) String() string {
 }
 
 func (a Address) MarshalJSON() ([]byte, error) {
+	if a == EmptyAddress {
+		return json.Marshal(nil)
```

Comment:
I'm not sure the design choice of empty address being encoded as null makes sense to be honest. 

If we use null as an actual value, we loose the ability to later use null to encode a lack of an address. 

Maybe I'm missing something, but the EmptyAddress could at least theoretically be the recipient of assets (e. g. to "burn" them, instead of dropping resources – not sure that makes sense).